src/compton/Compton_NWA.cc

```diff
@@ -67,38 +63,32 @@ Compton_NWA::Compton_NWA(const std::string &filehandle) {
 Compton_NWA::Compton_NWA(const std::string &filehandle,
                          const std::vector<double> &grp_bds, const size_t nxi) {
 
-  // make a group_data struct to pass to the lib builder:
-  multigroup::Group_data grp_data;
-  grp_data.group_bounds = grp_bds;
-  grp_data.lib_file = filehandle;
-  grp_data.n_leg = nxi;
-
-  // This is totally arbitrary, because I'm just assuming the user wants
-  // Planck-weighting...
-  grp_data.wt_func = multigroup::Weighting_function::PLANCK;
+  // Check input validity
+  Require(std::ifstream(filehandle).good());
+  Require(grp_bds.size() > 0);
 
-  // TODO: How do we actually want to handle the weighting function?
-  // Allow the user to pass it in? Make some intelligent decision at runtime?
   std::cout << "*********************************************************\n"
             << "WARNING! Building a multigroup library from scratch might\n"
             << " take a LOOOOOOONG time! (Don't say I didn't warn you.)  \n"
             << "*********************************************************\n"
             << std::endl;
 
-  // Make an mg lib builder, based on the group structure and pointwise lib:
-  multigroup_lib_builder MG_builder(grp_data);
+  // make a group_data struct to pass to the lib builder:
+  // TODO: How do we actually want to handle the weighting function?
+  // Allow the user to pass it in? Make some intelligent decision at runtime?
+  multigroup::Group_data grp_data = {multigroup::Library_type::EXISTING,
+                                     multigroup::Weighting_function::PLANCK,
+                                     filehandle, nxi, grp_bds};
+
+  // Transfer ownership of the grp_data struct directly to MG_builder, since
+  // it's a temporary which exists only for this purpose
+  multigroup_lib_builder MG_builder(std::move(grp_data));
 
   // build the library:
   MG_builder.build_library();
 
-  // package the data into a compton_data pointer:
-  std::shared_ptr<multigroup_compton_data> Cdata = MG_builder.package_data();
-
-  // Make sure the SP exists...
-  Ensure(Cdata);
-
   // initialize the electron temperature interpolator with the mg compton data
-  ei.reset(new etemp_interp(Cdata));
+  ei.reset(new etemp_interp(MG_builder.package_data()));
```

Comment:
This is basically the same thing that using etemp_interp(std::move(Cdata)) in (old) line 101 would've done, right? I'm keeping the argument to etemp_interp an rvalue by not storing it in Cdata, so I avoid a potential copy (though the compiler might've done away with it anyway).